bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -0,0 +1,301 @@
+    private static final String NO_ID = "NO_ID";
+    private final Logger logger = LoggerFactory.getLogger(AutomowerHandler.class);
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_AUTOMOWER);
+
+    private @Nullable AutomowerConfiguration config;
+
+    private AtomicReference<String> automowerId = new AtomicReference<String>(NO_ID);
+    private AtomicLong commandDuration = new AtomicLong(60); // 60min default
+    private long lastQueryTime = 0L;
+
+    private @Nullable ScheduledFuture<?> automowerPollingJob;
+    private long automowerPollingIntervalS = TimeUnit.HOURS.toSeconds(1);
+    private long maxQueryFrequencyNanos = TimeUnit.MINUTES.toNanos(1);
+
+    DateFormat df = DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.LONG,
+            new Locale(System.getProperty("user.language")));
+
+    public AutomowerHandler(Thing thing) {
+        super(thing);
+    }
+
+    class AutomowerPollingRunnable implements Runnable {
+
+        @Override
+        public void run() {
+            Bridge bridge = getBridge();
+            if (bridge != null && bridge.getStatus() == ThingStatus.ONLINE) {
+                updateAutomowerState();
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+
+        }
+
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refreshChannels(channelUID);
+        } else {
+            sendCommand(channelUID, command);
+        }
+    }
+
+    private void sendCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case CHANNEL_COMMAND_DURATION:
+                if (command instanceof DecimalType) {
+                    commandDuration.set(((DecimalType) command).longValue());
+                    updateState(CHANNEL_COMMAND_DURATION, new DecimalType(commandDuration.get()));
+                }
+                break;
+
+            case CHANNEL_COMMAND:
+                if (command instanceof StringType) {
+                    sendAutomowerCommand(command);
+                } else {
+                    logger.debug("Received command of type {} but StringType is expected.",
+                            command.getClass().getName());
+                }
+                break;
+
+        }
+    }
+
+    private void refreshChannels(ChannelUID channelUID) {
+        updateAutomowerState();
+    }
+
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            config = getConfigAs(AutomowerConfiguration.class);
+
+            final String configMowerId = config.getMowerId();
+            if (configMowerId != null) {
+                automowerId.set(configMowerId);
+
+                // bridgeHandler.registerLightStatusListener(this);
```

Comment:
Can this be removed?

Reply:
Yes, one question. Do you know, when the bridge would go down (state offline), would the things automatically become offline as well or would i have to listen for the state of the bridge and  adapt the things state accordingly?

Reply:
The child Things are set to offline with the detail BRIDGE_OFFLINE, when the bridge goes offline. Setting the child Things offline is not necessary.

Reply:
Perfect, thanks